plugins/CoreAdminHome/API.php

```diff
@@ -107,4 +114,37 @@ public function runCronArchiving()
         $archiver = new CronArchive();
         $archiver->main();
     }
+
+    /**
+     * Ensure the specified dates are valid.
+     * Store invalid date so we can log them
+     * @param array $dates
+     * @return Date[]
+     */
+    private function getDatesToInvalidateFromString($dates)
```

Comment:
Maybe this method could be moved somewhere else and made reusable? In the end you just give it some dates and you get back the invalid dates and the valid ones? Also some tests for this could be nice